internal/xds/translator/translator.go

```diff
@@ -463,6 +463,7 @@ func (t *Translator) addRouteToRouteConfig(
 			ea := &ExtraArgs{
 				metrics:       metrics,
 				http1Settings: httpListener.HTTP1,
+				ipFamily:      httpListener.IPFamily,
```

Comment:
this is incorrect, you are using the listener level `IPFamily` value in the cluster
you should be deriving this value using a new `IPFamily` field in 
https://github.com/envoyproxy/gateway/blame/48a0310082ceeaf4d3af17e6b6844c7d0fb5db0b/internal/ir/xds.go#L1297 instead and this should be set in the gateway api layer using the Backend Service / EndpointSlice info

Reply:
Thanks for the feedback! I’ve updated the code to use a new IPFamily field